worlds/mm_recomp/NormalRules.py

```diff
@@ -922,18 +921,18 @@ def get_location_rules(player, options):
             lambda state: can_plant_beans(state, player),
         "Stone Tower Inverted Outside Right Chest":
             lambda state: can_plant_beans(state, player),
-
+        #Stone Tower region access rules require all 'heavy' transforms and hookshot, hence why they're removed from the rules.
         # Stone Tower has 4 keys total
         "Stone Tower Temple Entrance Room Eye Switch Chest":
             lambda state: state.has("Progressive Bow", player),
         "Stone Tower Temple Entrance Room Lower Chest":
             lambda state: state.has("Small Key (Stone Tower)", player, 4) and state.has("Deku Mask", player) and can_use_light_arrows(state, player),
         "Stone Tower Temple Armos Room Back Chest":
-            lambda state: has_explosives(state, player) and has_mirror_shield(state, player) or can_use_light_arrows(state, player),
+            lambda state: has_explosives(state, player) and (has_mirror_shield(state, player) or can_use_light_arrows(state, player)),
```

Comment:
Explosives are only used for blowing up a hole to let light go through the room, which is then used to get rid of the sun block. If explosives are there as a means to defeat the Armoses, goron can already make quick work of them, which is needed in vanilla to get in the room in the first place.
```suggestion
            lambda state: (has_explosives(state, player) and has_mirror_shield(state, player)) or can_use_light_arrows(state, player),
```